rfc/2022/2022-07-14_Platform_Security_001_Service_Account_Guidance.md

```diff
@@ -0,0 +1,112 @@
+
+## Motivation
+
+The proposed Service Account Guidance would provide guidance on how service accounts should be requested, approved, managed and maintained.
+
+## Design
+
+The proposed [Service Account Guidance](https://vfs.atlassian.net/wiki/spaces/PSEC/pages/2222588013/Service+Account+Guidance) is a document in the PSEC Confluence page.
+
+Content below:
+
+#### What is a Service Account?
+A service account is a special type of shared/group privileged account used to execute applications, services, or other processes.  Service accounts are used to run automated or persistent processes and are intended to be used by applications, not interactively by people.
+
+#### What unique risks do Service Accounts pose?
+Service accounts are not tied back to a unique individual.   This makes tracing back actions performed by a service account to a specific individual a challenge. 
+
+Service account authenticators are often shared or known amongst a team.  This necessitates rotation of authenticators when a member of that team leaves the project.
+
+Service account authenticators are often stored within applications or code for use, and rotation of that authenticator can cause outage or downtime if not updated in all the applications, services, or other processes that rely upon its use.
+
+#### Platform Security Guidance for Service Accounts
+
+*Assigned Account Manager*
+
+Every service account should have an assigned Account Manager.  The Account Manager is a specific human individual with the overall responsibility for the service account but can delegate use of the account.  The Account Manager can be a team lead or member of the team that necessitates the use of the service account.  In the case of an Account Manager leaving, a replacement Account Manager from the team must be designated.
+
+*Service Account Identifiers*
+
+Service accounts should be named using a format that easily identifies it as a service account, includes the product name, managing team name, along with unique information about its intended purpose.  Generic names, individual accounts, or built-in administrative accounts should not be used.
+
+Recommended naming convention:  svc-${PRODUCT}-${TEAM}-${USAGE}
+
+**Examples:**
+
+Good: svc-platform-infra-nightly_cron_job
+
+Bad:  jwilson, service_account, admin
+
+*Required Approval*
+
+The Account Manager is responsible for initially requesting approval (link to request TBD) for the creation and usage of a service account.  Each service account should have a valid authorization approved by the ISO or designee (OCTO-DE VA employee).  In the case of an Account Manager leaving, the replacement Account Manager would need to submit a request for approval by the ISO or designee (OCTO-DE VA employee) designating themselves as the new Account Manager.
```

Comment:
How does the account manager request approval?

Reply:
It doesn't exist yet, but a github issue template for requesting a new service account.

Reply:
Resolved via
- https://github.com/department-of-veterans-affairs/va.gov-platform-architecture/pull/35/commits/7f9564c9dfb47dd0c13aec4f7389fb6e443dc949
- https://github.com/department-of-veterans-affairs/va.gov-platform-architecture/pull/35/commits/8eafc2032fd92b54c5d634165f1599eab8e944e9